pkg/api/controllers/samlconfig/samlconfig.go

```diff
@@ -0,0 +1,33 @@
+package samlconfig
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/rancher/rancher/pkg/auth/providers/saml"
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/rancher/types/config"
+)
+
+func Register(apiContext *config.ScaledContext) {
+	apiContext.Management.AuthConfigs("").AddHandler("authConfigController", sync)
+}
+
+func sync(key string, config *v3.AuthConfig) error {
+	samlConfig := &v3.SamlConfig{}
+	if key == "" || config == nil {
+		return nil
+	}
+
+	if samlConfStr, ok := config.Annotations["tempSamlConfigbyte"]; ok {
+		err := json.Unmarshal([]byte(samlConfStr), samlConfig)
+		if err != nil {
+			return fmt.Errorf("SAML controller: %v", err)
+		}
+	} else {
+		return nil
+	}
+
+	samlProvider := saml.Provider{}
+	return samlProvider.InitializeSPClient(samlConfig, config.Name)
```

Comment:
This function makes use of state on the samlProvider struct
Specifically:
```
	samlURL += s.Name
```
and
```
		Root.Get("PingLogin").Handler(samlspInstance.RequireAccount(http.HandlerFunc(s.HandleSamlPost)))
```

So, when you just create a new struct here and call the method, you are not going to get the desired result. The function `InitializeSPClient` should not be attached to any struct. It should be a stand alone function. It is going to have to "find" the correct provider from a map or something based on a name passed to it. Similar to how these functions work:
https://github.com/rancher/rancher/blob/33d900d485929bffbc907011ae47043de1cbf13b/pkg/auth/providers/providers.go#L27-L38

In fact, you should just use one of those. The only catch is that it returns a Provider interfact that will not a function exposed for configuring saml. So, you should create an interface specific to saml with the necessary reconfigure function on it and type switch it to your interface.